### PR TITLE
PLAT-808-4 Create SessionManager

### PR DESCRIPTION
**Pull request by daniel-klose**
- Create SessionManager to:
    - Get all sessions
    - Invalidate all sessions
    - Invalidate all non-admin session
  - Added manual logout logic to session overview
image

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionManager.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package com.softicar.platform.core.module.maintenance.session;
+
+import com.softicar.platform.ajax.document.AjaxDocument;
+import com.softicar.platform.ajax.framework.listener.AjaxSessionListener;
+import com.softicar.platform.core.module.CoreRoles;
+import com.softicar.platform.core.module.ajax.session.SofticarAjaxSession;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+
+public class SessionManager {
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionManager.java:10]: this class code looks very procedural, suggestion: make the methods non-static and add field `ServletContext context` initialized in the constructor?

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java [hunk 2]
```diff
@@ -1,12 +1,52 @@
...
+	private final IEmfDataTableDiv<HttpSession> table;
+
 	public SessionOverviewPageDiv() {
 
-		new EmfDataTableDivBuilder<>(new SessionOverviewTable()).buildAndAppendTo(this);
+		appendChild(
+			new DomActionBar(
+				new DomButton()//
+					.setClickCallback(SessionManager::invalidateAllSessions)
+					.setLabel(CoreI18n.LOGOUT_ALL)
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java:24]: intentionally no icon? maybe add `.setIcon(CoreImages.LOGOUT.getResource())`?

**Review by oliver-richers — DISMISSED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
+	private final IEmfDataTableDiv<HttpSession> table;
+
 	public SessionOverviewPageDiv() {
 
-		new EmfDataTableDivBuilder<>(new SessionOverviewTable()).buildAndAppendTo(this);
+		appendChild(
+			new DomActionBar(
+				new DomButton()//
+					.setClickCallback(new SessionManager()::invalidateAllSessions)
+					.setLabel(CoreI18n.LOGOUT_ALL)
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java:24]: I think we should relabel this to `TERMINATE_ALL_SESSIONS`.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java [hunk 4]
```diff
@@ -1,12 +1,53 @@
...
+					.setIcon(CoreImages.LOGOUT.getResource())
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java:25]: I think we don't need an icon here (much like in the screenshot which seems to be outdated).

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java [hunk 5]
```diff
@@ -1,12 +1,53 @@
...
+
+		@Override
+		public void buildCell(IEmfDataTableActionCell<HttpSession> cell, HttpSession session) {
+
+			cell
+				.appendChild(
+					new DomActionBar(
+						new DomButton()//
+							.setClickCallback(() -> invalidateSessionAndRefresh(session))
+							.setLabel(CoreI18n.LOGOUT)
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java:42]: I think we should use a title instead of a label, with text `TERMINATE SESSION`.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java [hunk 6]
```diff
@@ -1,12 +1,53 @@
...
+							.setIcon(CoreImages.LOGOUT.getResource())
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/maintenance/session/SessionOverviewPageDiv.java:43]: I think we should add a new icon, `TERMINATE_SESSION`, which looks like an X:

Either big...
close-big

...or small:
close-small

(feel free to download and use the SVG icons used in this message)

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java [hunk 7]
```diff
@@ -804,6 +804,10 @@ public interface CoreI18n extends EmfI18n {
 		.de("Tabellenname");
 	I18n0 TARGET_USER = new I18n0("Target User")//
 		.de("Ziel-Benutzer");
+	I18n0 TERMINATE_ALL_SESSIONS = new I18n0("Terminate all sessions")//
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java:807]:
```suggestion
	I18n0 TERMINATE_ALL_SESSIONS = new I18n0("Terminate All Sessions")//
```

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

**Review by alexander-schmidt — APPROVED** [review]

**Review by alexander-schmidt — APPROVED** [review]
